apps/teams-test-app/src/public/naa_childIframe.html

```diff
@@ -0,0 +1,219 @@
+        border: 1px solid #ccc;
+        padding: 2px;
+        height: 30px;
+      }
+
+      .default-button {
+        margin-left: 0;
+      }
+
+      .section-container h4 {
+        margin-top: 5px;
+        font-size: 16px;
+      }
+
+      .section-container {
+        border: 1px solid #ccc;
+        padding: 15px;
+        margin-bottom: 20px;
+        border-radius: 5px;
+      }
+    </style>
+  </head>
+  <body>
+    <div>
+      <div class="section-container">
+        <h4>NestedAppAuthBridge</h4>
+        <div class="input-row">
+          <input
+            id="input_childPayloadForBridge"
+            name="input_childPayloadForBridge"
+            type="text"
+            placeholder="Payload for NestedAppAuthBridge"
+          />
+          <input
+            id="button_sendPayloadToBridge"
+            name="button_sendPayloadToBridge"
+            type="button"
+            value="Send to NAA Bridge"
+            onClick="sendMessageToBridge()"
+          />
+        </div>
+        <input
+          id="button_defaultPayloadForBridge"
+          name="button_defaultPayloadForBridge"
+          type="button"
+          value="Default"
+          class="default-button"
+          onClick="setDefaultPayloadForBridge()"
+        />
+        <div id="text_bridge_response_for_child" class="response-box">No response yet</div>
+      </div>
+
+      <div class="section-container">
+        <h4>Top Window</h4>
+        <div class="input-row">
+          <input
+            id="input_childPayloadForTopWindow"
+            name="input_childPayloadForTopWindow"
+            type="text"
+            placeholder="Payload for Top Window"
+          />
+          <input
+            id="button_sendPayloadToTopWindow"
+            name="button_sendPayloadToTopWindow"
+            type="button"
+            value="Send to Top Window"
+            onClick="sendMessageToTopWindow()"
+          />
+        </div>
+        <input
+          id="button_defaultPayloadForTopWindow"
+          name="button_defaultPayloadForTopWindow"
+          type="button"
+          value="Default"
+          class="default-button"
+          onClick="setDefaultPayloadForTopWindow()"
+        />
+        <div id="text_topWindow_response_for_child" class="response-box">No response yet</div>
+      </div>
+    </div>
+
+    <script>
+      const targetOrigin = 'https://local.teams.office.com:8080';
+
+      // Utility function to display a response in a given element
+      const displayResponse = (element, message) => {
+        element.innerText = message;
+      };
+
+      // Predefined payloads
+      const defaultPayloadForBridge =
+        '{"messageType":"NestedAppAuthRequest","method":"GetToken","sendTime":1732269811006,"clientLibrary":"testOS","clientLibraryVersion":"1.0.0","requestId":"684352c2-7ab7-4def-b7e1-XXXXXXXXXXX","tokenParams":{"correlationId":"39dc85fe-9054-11ed-a1eb-XXXXXXXXXXX"}}';
+      const defaultPayloadForTopWindow =
```

Comment:
With this approach, if the NAA payload changes shape at all (name changes, property additions, etc.) how will someone know how to find and update these hardcoded strings? It might be easier to use TeamsJS directly the same way that the rest of the teams-test-app does

Reply:
Okay, I see that you've just taken the default payload from NestedAppAuthAPIs.tsx (after you pointed it out to me). Could you pass the payload from NestedAppAuthAPIs.tsx into here so that we don't have duplicate definitions?

Reply:
Thanks. As discussed offline, the payload is now passed from the parent to the child, eliminating the need for hardcoded payloads in the child.